### PR TITLE
New series landing page, new series single page and some ui

### PR DESCRIPTION
**Pull request by rocaiguina**
first I am not sure why/how, but here in the pull request it says I am missing the gitmodule for the INN repo, I have largo in my local installation though.

This pull request includes:
1. *new* series-landing.php - use existing template as a sample, copied necessary partials over to cpipr. It is used by selecting de calentamiento global template for Landing Pages.  The hero video, youtube video and abstract come from the cms.
2. *new* series-single.php based on the default template, it needs to be selected for posts.  It has a show next post from this series feature at the bottom.
3. we added some new styles and overrode some of the largo ones for now.  

series-single
landing

### CODE REVIEW

File: wp-content/themes/cpipr/footer.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		    if(counterHamburguer == false){
+		        $("#menu-header").slideDown("slow");
+		        counterHamburguer = true;
+		    }else{
+		        $("#menu-header").slideUp("slow");
+		        counterHamburguer = false;
+		    }
+		});
+	});
+</script>
```
Comment (benlk) [inline wp-content/themes/cpipr/footer.php]: You don't need to duplicate `footer.php` to add this. Add this snippet to the theme `functions.php` and then remove this `footer.php`

```php
/**
 * Counter Hamburger menu
 */
function cpipr_counter_hamburger_menu() {
    echo <<<DOC
<script>
	jQuery(document).ready(function($) {
		counterHamburguer = false;

		$('#menu-btn').click(function () {
		    if(counterHamburguer == false){
		        $("#menu-header").slideDown("slow");
		        counterHamburguer = true;
		    }else{
		        $("#menu-header").slideUp("slow");
		        counterHamburguer = false;
		    }
		});
	});
</script>
DOC;
}
add_action( 'wp_footer', 'cpipr_counter_hamburger_menu' );
```

File: wp-content/themes/cpipr/functions.php [hunk 2]
```diff
@@ -1,114 +1,152 @@
...
+ * LARGO_DEBUG defines whether or not to use minified assets
+ *
+ * Largo by default uses minified CSS and JavaScript files.
+ * set LARGO_DEBUG to TRUE to use unminified JavaScript files
+ * and unminified CSS files with sourcemaps to our LESS files.
+ *
+ * @since 0.5
+ */
+if ( ! defined( 'LARGO_DEBUG' ) )
+	define( 'LARGO_DEBUG', FALSE );
```
Comment (benlk) [inline wp-content/themes/cpipr/functions.php]: Remove this `LARGO_DEBUG` definition; it should only be set in wp-config.php.

File: wp-content/themes/cpipr/header.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+<!--[if lt IE 7]> <html <?php language_attributes(); ?> class="no-js ie6"> <![endif]-->
+<!--[if IE 7]>    <html <?php language_attributes(); ?> class="no-js ie7"> <![endif]-->
+<!--[if IE 8]>    <html <?php language_attributes(); ?> class="no-js ie8"> <![endif]-->
+<!--[if IE 9]>    <html <?php language_attributes(); ?> class="no-js ie9"> <![endif]-->
+<!--[if (gt IE 9)|!(IE)]><!--> <html <?php language_attributes(); ?> class="no-js"> <!--<![endif]-->
+<head>
+	<meta charset="<?php bloginfo( 'charset' ); ?>" />
+	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">
+	<link href="https://fonts.googleapis.com/css?family=Merriweather|Roboto|Roboto+Condensed" rel="stylesheet">
```
Comment (benlk) [inline wp-content/themes/cpipr/header.php]: You don't need to copy all of `header.php` into the theme just to add this line; look at how the function `cpipr_typekit` works.

Also, given that you're loading these fonts from the theme as well as from Google, you don't need to load these fonts from Google Fonts.

File: wp-content/themes/cpipr/header.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+		if ( SHOW_GLOBAL_NAV === TRUE ) {
+
+			/**
+			 * Fires before the Largo global navigation content.
+			 *
+			 * @since 0.4
+			 */
+			do_action( 'largo_before_global_nav' );
+
+			//get_template_part( 'partials/nav', 'global' );
```
Comment (benlk) [inline wp-content/themes/cpipr/header.php]: There are better ways to disable this menu.

You can:

1. copy `partials/nav-global.php` from Largo into this theme's `partials/` directory, and then make its contents empty except for:
    - an opening `<?php` tag and a comment explaining that the file is there to remove the global nav
    - the "skip to content" link
2. hide the whole thing with CSS

File: wp-content/themes/cpipr/header.php [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+			do_action( 'largo_after_global_nav' );
+
+		}
+	?>
+
+	<div id="page" class="hfeed clearfix">
+
+		<?php 
+			
+			//get_template_part( 'partials/nav', 'sticky' );
```
Comment (benlk) [inline wp-content/themes/cpipr/header.php]: See my comment about `partials/nav-global.php`. You should put a mostly-empty `partials/nav-sticky.php` in this theme, rather than comment out this line.

And since you're removing the templates that Largo's navigation javascript uses, you should use [`wp_dequeue_script( 'largo-navigation' )`](https://codex.wordpress.org/Function_Reference/wp_dequeue_script), to prevent those now-unnecessary scripts from being loaded. You could add that to the function `cpipr_styles` in `functions.php`, since at prirority 20 that function runs after [Largo enqueues the javascript](https://github.com/INN/largo/blob/master/inc/enqueue.php#L36-L41) with priority 10.

File: .gitmodules [hunk 6]
```diff
@@ -1,9 +1,6 @@
 [submodule "tools"]
 	path = tools
 	url = git@github.com:INN/deploy-tools.git
-[submodule "wp-content/themes/largo"]
-	path = wp-content/themes/largo
-	url = git@github.com:INN/Largo.git
```
Comment (benlk) [inline .gitmodules]: I can fix the largo git submodule removal if you [allow maintainers to edit your pull request](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/). This should be an option; let me know if it's not.

File: wp-content/themes/cpipr/less/_variables.less [hunk 7]
```diff
@@ -1,2 +1,69 @@
 @orange: #D45F34;
-@dk-orange: #BB522C;
\ No newline at end of file
+@dk-orange: #BB522C;
+
+@font-face {
+  font-family: 'GT america compressed black';
+  src: url('../fonts/gt-america-compressed-black-webfont.woff2') format('woff2'), url('../fonts/gt-america-compressed-black-webfont.woff') format('woff'), url('../fonts/gt-america-compressed-black-webfont.ttf') format('truetype'), url('../fonts/gt-america-compressed-black-webfont.svg#gt_americacompressed_black') format('svg');
```
Comment (benlk) [inline wp-content/themes/cpipr/less/_variables.less]: Pure curiosity on my part: does the `#gt_americacompressed_black` in these SVG font URLs have any effect on how the font loads?

File: wp-content/themes/cpipr/partials/content-series.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+$args = apply_filters( 'largo_content_partial_arguments', $args, get_queried_object() );
+
+extract( $args );
+
+$entry_classes = 'entry-content';
+
+$show_top_tag = largo_has_categories_or_tags();
+
+$authors = get_coauthors();
```
Comment (benlk) [inline wp-content/themes/cpipr/partials/content-series.php]: if the function `get_coauthors()` isn't defined, then the template breaks and the page fails to finish loading: 

screen shot 2018-04-09 at 1 19 52 pm

The parts of this template that are dependent upon the Co-Authors Plus plugin that defines `get_coauthors` should be wrapped in a conditional that checks if the function is defined before calling it, like so:

```php
if ( function_exists( 'get_coauthors' ) ) {
    $authors = get_coauthors();
} else {
    $authors = array();
    error_log( 'This theme depends upon Co-Authors Plus!' );
}
```

There are other places in the theme where `get_coauthors` and `coauthors_get_avatar` are called that should similarly be made safe.

**Review by benlk — CHANGES_REQUESTED** [review]
Summary: 
- I can fix the largo problem
- There's too many fonts being loaded now
- Changes to `header.php` and `footer.php` should be done differently, but that's not urgent
- Don't define `LARGO_DEBUG` in the child theme
- Don't copy the entire CF Taxonomy Landing plugin form Largo to this child theme
- Remove `package-lock.json`
- Make the theme safe if Co-Authors Plus is deleted

**Comment by benlk** [comment]
I have some concerns about this pull request, some of which I can fix and some of which are in your territory.

There are a number of files introduced in this pull request that are not referenced by the code, or that are unnecessary duplicates. Can you remove these, please?
- package-lock.json
- images/arrow-left.png
- images/arrow-right.png
- images/secondaryLogo.png
- inc/wp-taxonomy-landing/CHANGELOG.txt
- inc/wp-taxonomy-landing/README.txt


## Fonts

The `cpipr` theme loads Typekit fonts from the kit "dyj6ctr", using the function `cpipr_typekit` in functions.php. That kit is on INN's Typekit account, and loads only one font, Proxima Nova, at 90 kilobytes size. (Folks with access to INN's Typekit account can see this kit here: https://typekit.com/kit_editor/kits/dyj6ctr?popup=1 )

This pull request adds these fonts:

- GT America Compressed Black
- GT America Compressed Medium Italic
- GT America Compressed Regular
- Merriweather Light
- Merriweather Regular
- Roboto Bold
- Roboto Regular
- Roboto Condensed Bold
- Roboto Condensed Bold Italic
- Roboto Condensed Italic
- Roboto Condensed Regular

In addition, the `header.php` with this pull request adds a link to Google Fonts to pull in Merriweather, Roboto, and Roboto Condensed, with the URL https://fonts.googleapis.com/css?family=Merriweather|Roboto|Roboto+Condensed

On the CPI homepage, loading all those fonts takes 1.9 megabytes of data. That's an excessive amount of fonts to be loading on a page. Can you trim that back at all, by using system fonts for as much as possible?

Also, loading these fonts from the theme folders on periodismoinvestigativo.com may not be the best choice for speed of loading. Using Google Fonts means that periodismoinvestigativo.com can leverage the cached versions of those files from when readers' browsers have loaded those fonts from Google Fonts when they were used on other websites, which means they don't have to redownload those 1.9 MB of fonts when visiting periodismoinvestigativo.com with clear caches. If you don't want to use Google Fonts for privacy reasons, I understand that, but in that case you should remove the https://fonts.googleapis.com/css?family=Merriweather|Roboto|Roboto+Condensed request.

## Series landing pages

On the series landing page edit interface, I'm receiving a warning message in the server error log:

> PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in /Users/blk/sites/cpipr/wp-content/themes/cpipr/inc/wp-taxonomy-landing/functions/cftl-admin.php on line 382

But that looks to be a bug in the Largo functions that you copied over, so I'll make a note of it in Largo https://github.com/INN/largo/issues/1489 and move on.

You copied over all the cftl-tax-landing plugin files from Largo, but doing so broke the series landing pages. The options in the "Abstract" box saves, but nothing else saves. I believe this is because you changed the check `|| $_POST['post_type'] != 'cftl-tax-landing') {` to `|| $_POST['post_type'] != 'Series') {`  in the function `cftl_tax_landing_save_layout`.

You don't need to copy over _all_ of the CFTL Tax Landing Plugin's files.

Here's the changes in the local copy of each file from the Largo copy: 

<details>
<summary>
inc/wp-taxonomy-landing/functions/cftl-admin.php:
</summary>

```diff
13c13
< define('CFTL_SELF_DIR', get_template_directory_uri() . '/inc/wp-taxonomy-landing/');
---
> define('CFTL_SELF_DIR', get_stylesheet_directory_uri() . '/inc/wp-taxonomy-landing/');
```
If you don't copy all the files over from Largo, this is not necessary.


```diff
229a230,239
> 	//add the 'abstract' box
> 	add_meta_box(
> 		'cftl_tax_landing_abstract',
> 		__('Abstract', 'largo'),
> 		'cftl_tax_landing_abstract',
> 		'cftl-tax-landing',
> 		'normal',
> 		'default'
> 	);
> 
```
You can add the meta box through a separate function hooked on `add_meta_boxes` at priority 99, like the Largo function `cftl_tax_landing_add_extras_box` does. 

```diff
321c331
< 	<h4>Show Series Byline?</h4>
---
> 	<h4>Show Series Byline, what?</h4>
```
Was this meant to be left in?

```diff
519a530,604
> function cftl_tax_landing_abstract ( $post ) {
> 	wp_nonce_field( plugin_basename(__FILE__), 'cftl_tax_landing_abstract' );
> 	$fields = ($post->post_title) ? get_post_custom( $post->ID ) : cftl_field_defaults();
> 	?>
> 	
> 	<div class="form-field">
> 		<h4>abstract</h4>
> 		<div>
> 			<?php wp_editor( $fields['abstract_body'][0], 'abstract_body', array(
> 				'wpautop' => false,
> 				'textarea_rows' => 5,
> 				'teeny' => true,
> 			)); ?>
> 		</div>
> 	</div>
> 	<div class="form-field">
> 	<h4>abstract video youtube id</h4>
> 		<div>
> 			<input type="text" name="abstract_video" id="video" value="<?php if ( isset ( $fields['abstract_video'] ) ) echo $fields['abstract_video'][0]; ?>" />
> 		</div>
> 	</div>
> 	<div class="form-field">
> 	<h4>hero background video url</h4>
> 		<div>
> 			<input type="text" name="hero_video" id="video" value="<?php if ( isset ( $fields['hero_video'] ) ) echo $fields['hero_video'][0]; ?>" />
> 		</div>
> 	</div>
> <?php
> }
> 
> function cftl_tax_landing_save_abstract($post_id) {
> 	if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
> 		return;
> 	}
> 	if ( ! isset( $_POST['post_type'] )
> 		|| $_POST['post_type'] != 'cftl-tax-landing') {
> 		return;
> 	}
> 	if ( ! current_user_can( 'edit_post', $post_id ) ) {
> 		return;
> 	}
> 
> 	$abstract_fields = array(
> 		'abstract_video'          => 'sanitize_text_field',
> 		'abstract_body'           => 'wp_filter_post_kses',
> 		'hero_video'							=> 'esc_url',
> 	);
> 
> 	foreach ($abstract_fields as $field_name => $sanitize ) {
> 
> 		switch ( $sanitize ) {
> 			case 'bool':
> 				$safe_value = ! empty( $_POST[ $field_name ] ) ? true : false;
> 				break;
> 
> 			case 'sanitize_show':
> 
> 				$safe_value = array();
> 				foreach( array( 'image', 'excerpt', 'byline', 'tags' ) as $key ) {
> 					$safe_value[ $key ] = ! empty( $_POST[ $field_name ][ $key ] ) ? true : false;
> 				}
> 				break;
> 
> 			default:
> 				$safe_value = ! empty( $_POST[ $field_name ] ) ? call_user_func( $sanitize, $_POST[ $field_name ] ) : '';
> 				break;
> 
> 		}
> 
> 		update_post_meta( $post_id, $field_name, $safe_value );
> 	}
> }
> 
> add_action('save_post', 'cftl_tax_landing_save_abstract');
> 
```
This bit is excellent. I would prefer that you moved this and all of the new functionality to the landing page features in Largo go into a new file in the `cpipr` theme named `inc/cftl-tax-landing.php`, rather than copying the whole CFTL subdirectory from Largo into the child theme.

```diff
551c636
< 		|| $_POST['post_type'] != 'cftl-tax-landing') {
---
> 		|| $_POST['post_type'] != 'Series') {
```
That breaks a lot of things; please undo it.

```diff
675c760
<  * Helper function for loading in posts for custom post manager
---
>  * Helper function for loading in posts for custom post
```
This change isn't necessary.
</details>

<details>
<summary>
inc/wp-taxonomy-landing/functions/cftl-intercept-queries.php
</summary>

Nothing was changed in this file, relative to Largo.
</details>

<details>
<summary>
inc/wp-taxonomy-landing/functions/cftl-series-order.php
</summary>

Nothing was changed in this file, relative to Largo.
</details>

<details>
<summary>
inc/wp-taxonomy-landing/landing-layouts.png
</summary>

Nothing was changed in this file, relative to Largo.
</details>

<details>
<summary>
inc/wp-taxonomy-landing/series.css
</summary>

Nothing was changed in this file, relative to Largo.
</details>

<details>
<summary>
inc/wp-taxonomy-landing/series.js
</summary>

Nothing was changed in this file, relative to Largo.
</details>
<details>
<summary>
inc/wp-taxonomy-landing/taxonomy-landing.php
</summary>

Nothing was changed in this file, relative to Largo.
</details>

So in summary: the only changes to taxonomy landing page functionality are in inc/wp-taxonomy-landing/functions/cftl-admin.php, and those changes can be made without copying all of the taxonomy landing functionality and files from Largo. Instead, please put the new functionality you created into a separate file in `inc/`. For an example of how that works, see [`inc/series-landing.php` in the Chicago Reporter theme](https://github.com/INN/umbrella-chicagoreporter/tree/master/wp-content/themes/chicago-reporter/inc).

And then remove the `inc/wp-taxonomy-landing` folder from this theme.

**Comment by rocaiguina** [comment]
ok will get to it

**Comment by benlk** [comment]
Sounds great! Let us know when you're done and ready for another review.

**Comment by rocaiguina** [comment]
ok, I am only missing the inc/wp-taxonomy-landing issue.
I am still wrapping my head around the creation of the metaboxes and stuff.  It still kind of confusing.
should I make a file that just contains something like this
```
<?
largo_add_meta_box(
  'cftl_tax_landing_abstract',
  __('Abstract', 'largo'),
  'cftl_tax_landing_abstract',
  'cftl-tax-landing',
  'normal',
  'default'
);

function cftl_tax_landing_abstract ( $post ) {
  global $post;
	$values = get_post_custom( $post->ID );
	wp_nonce_field( 'largo_meta_box_nonce', 'meta_box_nonce' );
	?>
	
	<div class="form-field">
		<h4>abstract</h4>
		<div>
			<?php wp_editor( $values['abstract_body'][0], 'abstract_body', array(
				'wpautop' => false,
				'textarea_rows' => 5,
				'teeny' => true,
			)); ?>
		</div>
	</div>
	<div class="form-field">
	<h4>abstract video youtube id</h4>
		<div>
			<input type="text" name="abstract_video" id="abstract_video" value="<?php if ( isset ( $values['abstract_video'] ) ) echo $values['abstract_video'][0]; ?>" />
		</div>
	</div>
	<div class="form-field">
	<h4>hero background video url</h4>
		<div>
			<input type="text" name="hero_video" id="hero_video" value="<?php if ( isset ( $values['hero_video'] ) ) echo $values['hero_video'][0]; ?>" />
		</div>
	</div>
}
<?php

function cftl_tax_landing_save_abstract($post_id) {
	if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
		return;
	}
	if ( ! isset( $_POST['post_type'] )
		|| $_POST['post_type'] != 'cftl-tax-landing') {
		return;
	}
	if ( ! current_user_can( 'edit_post', $post_id ) ) {
		return;
	}

	$abstract_fields = array(
		'abstract_video'          => 'sanitize_text_field',
		'abstract_body'           => 'wp_filter_post_kses',
		'hero_video'							=> 'esc_url',
	);

	foreach ($abstract_fields as $field_name => $sanitize ) {
		switch ( $sanitize ) {
			case 'bool':
				$safe_value = ! empty( $_POST[ $field_name ] ) ? true : false;
				break;
			case 'sanitize_show':
				$safe_value = array();
				foreach( array( 'image', 'excerpt', 'byline', 'tags' ) as $key ) {
					$safe_value[ $key ] = ! empty( $_POST[ $field_name ][ $key ] ) ? true : false;
				}
				break;
			default:
				$safe_value = ! empty( $_POST[ $field_name ] ) ? call_user_func( $sanitize, $_POST[ $field_name ] ) : '';
				break;
		}
		update_post_meta( $post_id, $field_name, $safe_value );
	}
}

add_action('save_post', 'cftl_tax_landing_save_abstract');

```

**Comment by rocaiguina** [comment]
@benlk hey, if you are still working could you please give me a hand trying to figure out how to go about creating the metabox and properly saving it from the functions file?  I am trying to figure it out following the tutorials you sent but can't seem to apply it properly.  They are trying to launch the series tomorrow.

**Comment by benlk** [comment]
Taking a look

**Comment by benlk** [comment]
What you had in the comment was pretty close; the changes I made to it are reflected in the file added in the commit https://github.com/INN/umbrella-cpipr/pull/9/commits/5f5104cd4c2b443c23c74c21184253d2a74c011e

The key things were:

- `<?` is not a valid opening PHP tag
- there were some misbalanced {} in the function cftl_tax_landing_abstract that meant that the function never really ended, I think.

I've removed the copy of `inc/wp-taxonomy-landing` from the child theme, and now the usual fields on the taxonomy landing page work. Moving on to the code review of the changes.

**Comment by rocaiguina** [comment]
one thing, I don't see the new fields on the series landing admin view. I am checking it out

**Comment by rocaiguina** [comment]
just commited changing the call to largo_add_meta_box()  to just add_meta_box(), and now it's working.

**Comment by benlk** [comment]
That last commit fixes an issue with the series landing template that was causing posts loaded via the Load More Posts/Cargar Más Historias button to have left margins.

per-page font loads are way down, which is good.

**Comment by benlk** [comment]
All review comments are outdated and there have been some tangible improvements.

I'm still kinda queasy about having so many fonts, but on second inspection the *total* page load was in the low megabytes and the individual fonts loaded are down below a hundred kilobytes.

I'm gonna merge and deploy tomorrow morning, because deploying by hand is a bad idea and so is — at this time of night — tweaking the deploy scripts to be in accordance with https://github.com/INN/deploy-tools/issues/61

**Comment by benlk** [comment]
This is now deployed.
